platform-tests/example-apps/http_tester/main.go

```diff
@@ -0,0 +1,159 @@
+package main
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	delay   = 250
+	letters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+	BYTE     = int64(1)
+	KILOBYTE = 1024 * BYTE
+	MEGABYTE = 1024 * KILOBYTE
+	GIGABYTE = 1024 * MEGABYTE
+	TERABYTE = 1024 * GIGABYTE
+)
+
+func main() {
+	addr := ":" + os.Getenv("PORT")
+	fmt.Println("Listening on", addr)
+	http.HandleFunc("/", root)
+	http.HandleFunc("/print-headers", printHeaders)
+	http.HandleFunc("/body-size", bodySize)
+	http.HandleFunc("/header-size", headerSize)
+	http.HandleFunc("/big-header", bigHeader)
+	http.HandleFunc("/egress", egress)
+	http.HandleFunc("/slow-response", slowResponse)
+	http.HandleFunc("/slow-request", slowRequest)
+	http.HandleFunc("/long-url", longURL)
+	err := http.ListenAndServe(addr, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func root(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte("OK"))
+}
+
+func printHeaders(w http.ResponseWriter, r *http.Request) {
+	jsonData, err := json.Marshal(r.Header)
+	if err != nil {
+		http.Error(w, fmt.Sprint(err), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(jsonData)
+}
+
+func bodySize(w http.ResponseWriter, r *http.Request) {
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprint(err), http.StatusInternalServerError)
+		return
+	}
+	w.Write([]byte(fmt.Sprintf("%d", len(body))))
+}
+
+func headerSize(w http.ResponseWriter, r *http.Request) {
+	header := r.Header.Get("test-header")
+	w.Write([]byte(fmt.Sprintf("%d", len(header))))
+}
+
+func bigHeader(w http.ResponseWriter, r *http.Request) {
+
+	var headerSize = r.URL.Query().Get("size")
+	headerInt, err := strconv.Atoi(headerSize)
+	if err != nil {
+		http.Error(w, fmt.Sprint(err), http.StatusInternalServerError)
+		return
+	}
+	rand.Seed(time.Now().UnixNano())
+
+	w.Header().Set("test-header", randStringBytes(headerInt*int(KILOBYTE)))
+}
+
+func egress(w http.ResponseWriter, r *http.Request) {
+
+	domain := r.URL.Query().Get("domain")
+
+	var responseMessage string
+	var responseCode int
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```

Comment:
This was removed and added back without comment in:

https://github.com/alphagov/paas-cf/pull/931/files/09e8efbcaa31d8aae76cde7f66d7fe3aa0ac613f#diff-2c8d4ffb16bb71b88fd7b89690145eed

I don't think the solutions are easy though because `config.SkipSSLValidation` is set to false in all environments. Let's leave it for now.